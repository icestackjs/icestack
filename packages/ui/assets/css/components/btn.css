.btn-primary {
  @apply border-primary bg-primary text-primary-content outline-primary;
}
.btn-success {
  @apply bg-success;
}
.btn-warning {
  @apply bg-warning;
}
.btn-error {
  @apply bg-error;
}