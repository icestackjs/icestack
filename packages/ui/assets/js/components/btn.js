module.exports = {
  ".btn-primary": {
    "@apply border-primary bg-primary text-primary-content outline-primary": true
  },
  ".btn-success": {
    "@apply bg-success": true
  },
  ".btn-warning": {
    "@apply bg-warning": true
  },
  ".btn-error": {
    "@apply bg-error": true
  }
}