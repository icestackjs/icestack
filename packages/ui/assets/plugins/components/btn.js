const _plugin = require("tailwindcss/plugin");
const returnSelfNoop = x => x;
const css2TwPlugin = _plugin.withOptions(function (_options = {}) {
  const {
    withOptionsWalkCSSRuleObject = returnSelfNoop
  } = _options;
  return function ({
    addBase: addBase,
    addComponents: addComponents,
    addUtilities: addUtilities,
    theme: theme,
    addVariant: addVariant,
    config: config,
    corePlugins: corePlugins,
    e: e,
    matchComponents: matchComponents,
    matchUtilities: matchUtilities,
    matchVariant: matchVariant
  }) {
    const _baseCss = withOptionsWalkCSSRuleObject({}, "base");
    addBase(_baseCss);
    const _componentsCss = withOptionsWalkCSSRuleObject({
      ".btn-primary": {
        "--tw-border-opacity": "1",
        "border-color": "rgb(var(--primary) / var(--tw-border-opacity))",
        "--tw-bg-opacity": "1",
        "background-color": "rgb(var(--primary) / var(--tw-bg-opacity))",
        "--tw-text-opacity": "1",
        "color": "rgb(var(--primary-content) / var(--tw-text-opacity))",
        "outline-color": "rgb(var(--primary) / 1)"
      },
      ".btn-success": {
        "--tw-bg-opacity": "1",
        "background-color": "rgb(var(--success) / var(--tw-bg-opacity))"
      },
      ".btn-warning": {
        "--tw-bg-opacity": "1",
        "background-color": "rgb(var(--warning) / var(--tw-bg-opacity))"
      },
      ".btn-error": {
        "--tw-bg-opacity": "1",
        "background-color": "rgb(var(--error) / var(--tw-bg-opacity))"
      }
    }, "components");
    addComponents(_componentsCss);
    const _utilitiesCss = withOptionsWalkCSSRuleObject({}, "utilities");
    addUtilities(_utilitiesCss);
  };
}, function (_options) {
  return {};
});
module.exports = css2TwPlugin;