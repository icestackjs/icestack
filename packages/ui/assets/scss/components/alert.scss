.alert {
  @apply rounded-box text-base-content border-base-200 border p-4;

  &-info {
    @apply text-info-content border-info/20 bg-info;
  }
  &-success {
    @apply text-success-content border-success/20 bg-success;
  }
  &-warning {
    @apply text-warning-content border-warning/20 bg-warning;
  }
  &-error {
    @apply text-error-content border-error/20 bg-error;
  }
}
