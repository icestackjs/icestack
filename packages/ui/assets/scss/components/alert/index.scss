@use "../../utils/index.scss" as *;
@use "sass:map";
$TypeMap: inject("alert.colors");
$DefaultMap: inject("alert.defaults.styled");

.alert {
  @include resolve(map.get($DefaultMap, "default"));

  @each $typeName, $map in $TypeMap {
    &#{getSelector($typeName)} {
      @include resolve(map.get($map, "default"));
    }
  }
}
