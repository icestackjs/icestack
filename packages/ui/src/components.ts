export { componentNames as names, removeDefaultComponents, schemaMap } from '@icestack/preset-default/components'
