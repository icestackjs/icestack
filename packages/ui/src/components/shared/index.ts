export type * from '@icestack/shared'
export * from './utils'
