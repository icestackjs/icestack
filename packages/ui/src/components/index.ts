import { GetSchemaFn, preprocessCssInJs } from './shared'
import collapse from './collapse'
import join from './join'
import indicator from './indicator'
import divider from './divider'
import stack from './stack'
import tab from './tab'
import avatar from '@/components/avatar'
import button from '@/components/button'
import alert from '@/components/alert'
import badge from '@/components/badge'
import chat from '@/components/chat'
import checkbox from '@/components/checkbox'
import input from '@/components/input'
import link from '@/components/link'
import progress from '@/components/progress'
import textarea from '@/components/textarea'
import toggle from '@/components/toggle'
import select from '@/components/select'
import radio from '@/components/radio'
import range from '@/components/range'
import loading from '@/components/loading'
import mask from '@/components/mask'
import form from '@/components/form'
import table from '@/components/table'
import skeleton from '@/components/skeleton'
import radialProgress from '@/components/radialProgress'
import countdown from '@/components/countdown'
import diff from '@/components/diff'
import kbd from '@/components/kbd'
import tooltip from '@/components/tooltip'
import toast from '@/components/toast'
import steps from '@/components/steps'
import dropdown from '@/components/dropdown'
import swap from '@/components/swap'
import card from '@/components/card'
import carousel from '@/components/carousel'
import stat from '@/components/stat'
import timeline from '@/components/timeline'
import breadcrumbs from '@/components/breadcrumbs'
import menu from '@/components/menu'
import bottomNavigation from '@/components/bottom-navigation'
import navbar from '@/components/navbar'
import fileInput from '@/components/file-input'
const schemaMap = {
  alert,
  avatar,
  button,
  badge,
  chat,
  checkbox,
  input,
  link,
  progress,
  textarea,
  toggle,
  select,
  radio,
  range,
  loading,
  mask,
  form,
  table,
  // tabs,
  skeleton,
  'radial-progress': radialProgress,
  countdown,
  diff,
  kbd,
  tooltip,
  toast,
  steps,
  collapse,
  join,
  indicator,
  divider,
  stack,
  tab,
  dropdown,
  swap,
  card,
  carousel,
  stat,
  timeline,
  breadcrumbs,
  menu,
  'bottom-navigation': bottomNavigation,
  navbar,
  'file-input': fileInput
} as const

const componentNames = Object.keys(schemaMap) as (keyof typeof schemaMap)[] // as unknown as keyof typeof _schemaMap

const resolvedSchemaMap = {} as Record<string, { schema: GetSchemaFn }>
for (const componentName of componentNames) {
  const o = schemaMap[componentName as keyof typeof schemaMap].schema
  resolvedSchemaMap[componentName] = {
    schema: (...args) => {
      const { defaults, selector } = o(...args)
      return {
        selector,
        defaults: preprocessCssInJs(defaults)
      }
    }
  }
}

// const names = Object.keys(resolvedSchemaMap) as (keyof typeof schemaMap)[]

const removeDefaultComponents = componentNames.reduce<Record<string, false>>((acc, cur) => {
  acc[cur] = false
  return acc
}, {})
export { componentNames as names, schemaMap, resolvedSchemaMap, removeDefaultComponents }

export { expandTypes, getSelector, compressCssSelector, preprocessCssInJs, transformCss2Js } from './shared'

export type { GetSchemaFn } from './shared'
