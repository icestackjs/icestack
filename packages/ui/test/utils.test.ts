import { compressCssSelector, applyStringToArray } from '@/components/shared'

describe('utils', () => {
  it('compressCssSelector case 0', () => {
    expect(compressCssSelector('h1, h2, h3')).toBe('h1,h2,h3')
    expect(compressCssSelector('>   div')).toBe('>div')
    expect(compressCssSelector('span  div')).toBe('span div')
    expect(compressCssSelector('span  ,div')).toBe('span,div')
  })

  it('applyStringToArray case 0', () => {
    expect(
      applyStringToArray({
        '.x': {
          css: {
            color: 'red'
          },
          apply: 'bg-gray-100 text-blue',
          '&-x   .c  >  .b': {
            css: {
              color: 'red'
            },
            apply: 'bg-gray-100 text-blue'
          }
        }
      })
    ).toMatchSnapshot()
  })

  it('applyStringToArray case 1', () => {
    expect(
      applyStringToArray({
        '.x .x': {
          css: {
            color: 'red'
          },
          apply: 'bg-gray-100 text-blue'
        },
        '.x  .x': {
          css: {
            color: 'blue'
          },
          apply: 'bg-gray-200 text-red'
        },
        '.x    .x': {
          css: {
            color: 'yello'
          },
          apply: 'bg-gray-300 text-blue'
        }
      })
    ).toMatchSnapshot()
  })

  it('applyStringToArray case 2', () => {
    const res0 = applyStringToArray({
      '.x .x': {
        css: {
          color: 'red'
        },
        apply: 'bg-gray-100 text-blue'
      },
      '.x  .x': {
        css: {
          color: 'blue'
        },
        apply: 'bg-gray-200 text-red'
      },
      '.x    .x': {
        css: {
          color: 'yello'
        },
        apply: 'bg-gray-300 text-blue'
      }
    })
    expect(res0).toEqual(applyStringToArray(res0))
  })
})
