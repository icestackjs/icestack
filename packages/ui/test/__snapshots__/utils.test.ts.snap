// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`utils > applyStringToArray case 0 1`] = `
{
  ".x": {
    "&-x .c>.b": {
      "apply": [
        "bg-gray-100",
        "text-blue",
      ],
      "css": {
        "color": "red",
      },
    },
    "apply": [
      "bg-gray-100",
      "text-blue",
    ],
    "css": {
      "color": "red",
    },
  },
}
`;

exports[`utils > applyStringToArray case 1 1`] = `
{
  ".x .x": {
    "apply": [
      "bg-gray-300",
      "text-blue",
    ],
    "css": {
      "color": "yello",
    },
  },
}
`;
